### ISSUE TITLE
azurerm_windows_web_app site_config.application_stack is cleared while applying

### ISSUE THREAD

**Issue by peterprumbach-telekom** [hashicorp/terraform-provider-azurerm#18636]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.1

### AzureRM Provider Version

0.39.0

### Affected Resource(s)/Data Source(s)

azurerm_windows_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_windows_web_app" "webapp" {
  name                       = var.resource_name
  location                   = var.location
  resource_group_name        = var.rg_name
  service_plan_id            = var.app_service_plan_id

  site_config {
    application_stack {
      java_version           = "1.8"
      java_container         = "TOMCAT"
      java_container_version = "8.5.72"
      current_stack          = "java"
      dotnet_version         = "v4.0"
    }
   ...
}
```

### Debug Output/Panic Output

```shell
10:41:22        ~ site_config {
10:41:22              # (22 unchanged attributes hidden)
10:41:22  
10:41:22            ~ application_stack {
10:41:22                + current_stack          = "java"
10:41:22                + java_container         = "TOMCAT"
10:41:22                + java_container_version = "8.5.72"
10:41:22                + java_version           = "1.8"
10:41:22                  # (1 unchanged attribute hidden)
10:41:22              }
10:41:22          }
```

### Expected Behaviour

No change should be made, as current_stack corresponds to the Web App settings.

### Actual Behaviour

Terraform removes all attributes within current_stack and deletes the configuration (there is no terraform output showing the removal of the arguments). If you apply again with the same configuration, the attributes are added again.

Screenshot from the Activity Log in Azure:
1st "apply":
![grafik](https://user-images.githubusercontent.com/113669010/194021851-9768feea-d59d-4b43-a066-aef3b1015345.png)

2nd "apply:
![grafik](https://user-images.githubusercontent.com/113669010/194021990-5c1f3bbf-d531-4e77-b77b-0802526c1f1c.png)

also, same behaviour on the metadata argument:
![grafik](https://user-images.githubusercontent.com/113669010/194022129-640b7e6d-4a6f-4142-9440-35882b00e29a.png)

### Important Factoids

Azure GermanyWestCentral

**Comment by peterprumbach-telekom** [comment]
#18568

**Comment by rcskosir** [comment]
@peterprumbach-telekom Thank you for opening this issue, the PR you mentioned (#18568) has merged. Therefore, I am going to mark this issue as closed.
Per the PR:
This functionality has been released in [v3.34.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.34.0/CHANGELOG.md). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new/choose) following the template. Thank you!
